Initial code for exit strategies module

whymath opened the pull request:
V0 code for exit strategies module, returning dictionary of lockdown vector dataframes

shubham1217 commented:
hey @whymath the CI failed mostly due to linting issues and import errors. Import is due to importing from source instead of `help_project.src`. Please retag me to review once CI has passed.

whymath commented:
@shubham1217 Build was successful - https://travis-ci.org/github/Auquan/help_project/builds/695333665

List of changes after initial commit:
1) Changed exitstrategies output lockdown vector to dataframe
2) Refactored exitstrategies module for naming conventions
3) pylint - Suppressed C0111 (Missing docstring) and R0201 (Method could be a function)
4) pylint - Suppressing W0703 (Catching too general exception Exception)

shubham1217 commented on test/exitstrategies/test_main.py:
```diff
@@ -0,0 +1,22 @@
+Test for dummy lockdown vector
+"""
+
+from help_project.src.exitstrategies.interface import ExitStrategies
```
instead of test_main can we raname to test_interface as we should plan on adding a test/commit hook that cheks appropriate test files have been added automatically. At that point having `test_filename` to check against will be helpful!

whymath replied:
This is done

whymath commented on .travis.yml:
```diff
@@ -6,5 +6,5 @@ install:
 # command to run tests
 script:
-  - pylint -d C0301,W0511 src/ test/
+  - pylint -d C0301,W0511,C0111,R0201,W0703 src/ test/
```
Removed the try-catch from the code and removed W0703 (Catching too general exception Exception) from list of pylint exceptions

whymath commented on src/exitstrategies/data/S1_2020_05_29_NZL.csv:
```diff
@@ -0,0 +1,31 @@
+category, focus_area, pol_pct, impl_pct
```
I have retained this without snake_case for the time-being, will create new PR once we finalize the format internally

whymath commented on src/exitstrategies/elt/dataextractor.py:
```diff
@@ -0,0 +1,53 @@
+"""
```
This is done

whymath commented:
@shubham1217 Build was successful - https://travis-ci.org/github/Auquan/help_project/builds/696129497

List of changes after previous review:
- Checked-in V0 code of exitstrategies module, with time-series lockdown vector for India
- Changed exitstrategies output lockdown vector to dictionary of dataframes
- Minor refactoring for naming conventions
- pylint - Undid suppressing of W0703 (Catching too general exception Exception)
- pylint - Suppressed W0640 (Cell variable defined in loop)

harshidw commented on src/exitstrategies/data_elt.py:
```diff
@@ -0,0 +1,80 @@
+"""
```
`exitstrategies` in snake case

whymath replied:
This is TBD at the next team meet, will be retaining current naming as discussed

harshidw commented on .travis.yml:
```diff
@@ -6,5 +6,5 @@ install:
 # command to run tests
 script:
-  - pylint -d C0301,W0511 src/ test/
+  - pylint -d C0301,W0511,C0111,R0201,W0640 src/ test/
```
I'm fine with the exceptions `C0111` and `R0201`, but I think `W0640` shouldn't be ignored for all the cases. 
The params passed to the lambda function might change when the function is actually invoked in case it was declared inside a loop. This thread explains better: https://stackoverflow.com/questions/938429/scope-of-lambda-functions-and-their-parameters/938493#938493

If the warning is a false positive in your case then you can add the linting exception to your scope instead of doing it globally.

whymath replied:
Replaced lambda function with list comprehension, thanks for the suggestion @harshidw !

whymath commented:
@shubham1217 @harshidw Build was successful - https://travis-ci.org/github/Auquan/help_project/builds/696409645

List of changes after previous review:
- pylint - Undid suppressing W0640 (Cell variable defined in loop)
